curiosrc/seal/task_submit_precommit.go

```diff
@@ -127,37 +127,49 @@ func (s *SubmitPrecommitTask) Do(taskID harmonytask.TaskID, stillOwned func() bo
 			F05DealID         int64 `db:"f05_deal_id"`
 			F05DealEndEpoch   int64 `db:"f05_deal_end_epoch"`
 			F05DealStartEpoch int64 `db:"f05_deal_start_epoch"`
+
+			DDODealStartEpoch int64 `db:"direct_start_epoch"`
+			DDODealEndEpoch   int64 `db:"direct_end_epoch"`
 		}
 
 		err = s.db.Select(ctx, &pieces, `
-		SELECT piece_index, piece_cid, piece_size, f05_deal_id, f05_deal_end_epoch, f05_deal_start_epoch
+               SELECT piece_index,
+                      piece_cid,
+                      piece_size,
+                      f05_deal_id,
+                      COALESCE(f05_deal_end_epoch, 0) AS  f05_deal_end_epoch,
+                      COALESCE(f05_deal_start_epoch, 0) AS f05_deal_start_epoch,
+                      COALESCE(direct_start_epoch, 0) AS direct_start_epoch,
+                      COALESCE(direct_end_epoch, 0) AS direct_end_epoch
```

Comment:
Nit, maybe coalesce as `COALESCE(f05_deal_end_epoch, direct_end_epoch, 0) AS deal_end`, should clean up a lot of logic below